lib/fs/fakefs.go

```diff
@@ -221,6 +237,10 @@ func (fs *fakefs) create(name string) (*fakeEntry, error) {
 		return entry, nil
 	}
 
+	if fs.insens {
+		name = UnicodeLowercase(name)
```

Comment:
This means the name of the entry will also be lowercase. Don't we want the name on creation to be as given in the argument and just the map key to be lowercase?